Autostory base

jw098 opened the pull request:
The Autostory.h file, on its own.
#484 should be merged first.

Mysticial commented on SerialPrograms/Source/PokemonSV/Programs/AutoStory/PokemonSV_AutoStory.cpp:
```diff
@@ -0,0 +1,648 @@
+            continue;
+        }
+        
+        std::string leading_zero = checkpoint < 10 ? "0" : "";
```
A cleaner way to do this is:
```
const size_t DIGITS = 3;
std::string number = std::to_string(checkpoint);
if (number.size() < DIGITS){
    number = std::string(DIGITS - number.size(), '0') + number;
}
```
which will work for any # of digits.

Mysticial commented on SerialPrograms/Source/PokemonSV/Programs/AutoStory/PokemonSV_AutoStory.cpp:
```diff
@@ -0,0 +1,648 @@
+    }
+}
+
+std::string AutoStory::end_segment_description(){
```
There's one of these for every segment right? How many segments are we expecting? Like 50+ of them?

I'm wondering if this should be encapsulated into the the segments themselves. Like we define an abstract class representing a segment, then we have each segment inherit it and implement all the required stuff such as these start/end messages and the actual logic.

This approach will require refactoring the segment dropdown away from EnumDropdown to something more generic like the StringSelect options which we use in places like Pokemon name dropdowns. These are built from arrays rather than compile-time lists of enums.

jw098 commented:
I refactored Autostory so that each segment is it's own class. This let's us run the segments from the list, based on index, instead of a large switch statement.

We want our autostory to have a break after the tutorial, to allow the user to send in the level 100 Gardevoir as the lead. This is why I added the EnumDropdownOption STORY_SECTION, to toggle between the Tutorial and Main story segments. But since we're trying to PR the code in stages, the main story segments haven't been PR'ed yet. This is why StringSelectOption STARTPOINT_MAINSTORY and ENDPOINT_MAINSTORY are commented out.
